### PR TITLE
test(core, platform): unit test fixes without ivy enabled

### LINKED ISSUE #5742: bug: core, platform - unit tests are failing if ivy is disabled
#### Is this a bug, enhancement, or feature request?
Bug
#### Briefly describe your proposal.
If diaable ivy in tsconfig.spec.json, then majority of the tests will fail with different errors like `Type ApprovalFlowAddNodeComponent is part of the declarations of 2 modules`, etc. https://gist.github.com/N1XUS/19c6e242f5eb7bab762dc2d655120e01
#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
0.31
#### If this is a bug, please provide steps for reproducing it.
1. open libs/platform/tsconfig.spec.ts
2. add ```    "angularCompilerOptions": {
      "enableIvy": false
    }``` to the config
3. Run `ng test platform`

### PR DESCRIPTION
**Pull request by platon-rov**
#### Please provide a link to the associated issue.

Closes #5742, #4206, #4208, #4210, #4196.

#### Please provide a brief summary of this pull request.

Unit tests improvements, error fixes.

To test:
* Add `"angularCompilerOptions": { "enableIvy": false }` to the `tsconfig.base.json`.
* Run `npm run test` & `npm run test:platform`
* See working tests :)

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [n/a] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [n/a] Documentation Examples
- [n/a] Stackblitz works for all examples

### CODE REVIEW

**Comment by dimamarksman** [comment]
One failed

image

**Comment by InnaAtanasova** [comment]
Are you sure the PR title should be `test`? You are changing component/directive files, too.

**Comment by platon-rov** [comment]
> Are you sure the PR title should be `test`? You are changing component/directive files, too.

Quite a small refactoring.

File: libs/platform/src/lib/components/form/step-input/base.step-input.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ export abstract class StepInputComponent extends BaseInput implements OnInit {
     }
 
     /** @hidden */
-    _contentDensity: ContentDensity = this.config.contentDensity;
+    _contentDensity = this.config.contentDensity;
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/form/step-input/base.step-input.ts:159]: why did you remove this?

File: libs/core/src/lib/slider/slider.component.ts [hunk 2]
```diff
@@ -381,7 +371,7 @@ export class SliderComponent implements OnInit, OnChanges, OnDestroy, ControlVal
     }
 
     /** @hidden */
-    getValuenow(position: number | number[], sliderValueTarget: SliderValueTargets): string | number {
+    getValueNow(position: number | number[], sliderValueTarget: SliderValueTargets): string | number {
```
Comment (dimamarksman) [inline libs/core/src/lib/slider/slider.component.ts:374]: I'm not sure we are allowed to refactor components in the scope of "unit tests with ivy disabled " issue.
I would suggest creating a separate issue for that.

File: libs/platform/src/lib/components/form/step-input/base.step-input.ts (hunk 1 above)
Comment (platon-rov) [reply libs/platform/src/lib/components/form/step-input/base.step-input.ts:159]: Redundant. But okay, if we're not permitted to change not spec files in spec pr - I will revert this.

**Review by kavya-b — APPROVED** [review]
Great work! Passes both core and platform tests, with and without ivy.

P.S: I also linked a few other issues(related to skipped tests) yesterday. Could you assign yourself to those and add them to the PR description too if they look like the correct issues?
image

**Review by dimamarksman — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]
